src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/CompletableHelper.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface CompletableHelper {
+    static CompletableFuture<Void> completeRunnable(Runnable runnable) {
+        /*
+         * NOTE: this is done here instead of offloading to the forkJoin pool "CompletableFuture.runAsync()"
+         * because we probably should think about a threading model before doing this.
+         * Maybe consider using Netflix's RxJava too (much simpler than CompletableFuture).
+         */
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+        try {
+            runnable.run();
+            completableFuture.complete(null);
+        } catch (Throwable t) {
```

Comment:
it the design pattern here to catch throwable?
normally you should only catch exception, and leave errors alone

Reply:
I'm not sold on that.

I personally think it would be a mistake to just catch Exceptions here ... think about the blocking call .get() this might never return if we only catch exceptions and an error occured. 
The other point is that the state of the failed execution would not be caught and presented to thread checking the completable.

Reply:
anyone else have thoughts on this?

Reply:
i personally still don't support whole block: assign future, try/catch with separate suppose "long" operation and separately completing future. it's just.. why?
java8 introduced `CompletableFuture` api for a reason and very much reliable. the main recommendation is to use your own executioner so to not fork from common pool - all should be good. even the same thing was done with service bus library's latest version jump introduced in [bulk scan processor](https://github.com/hmcts/bulk-scan-processor/pull/189)

Reply:
on top of that, future will do exactly the same and catch `Throwable` 😉